### PR TITLE
Add 'clampValue' to DecayAnimationConfig

### PR DESCRIPTION
**Pull request by aleclarson**
This allows the user to specify a value at which a `DecayAnimation` should automatically stop. Similar to the `overshootClamping` option for `SpringAnimation`.

:+1:

### CODE REVIEW

**Comment by ide** [comment]
cc @sahrens this looks convenient but is also addressed already with Interpolations (I think?)

**Comment by aleclarson** [comment]
There are times when you want to clamp a value for a specific animation, and not all the time.

File: Libraries/Animated/Animated.js [hunk 1]
```diff
@@ -291,6 +301,13 @@ class DecayAnimation extends Animation {
     window.cancelAnimationFrame(this._animationFrame);
     this.__debouncedOnEnd({finished: false});
   }
+
+  _tryClamping(
```
Comment (sahrens) [inline Libraries/Animated/Animated.js]: `_shouldClamp` would be a better name I think.

**Comment by sahrens** [comment]
Hmm yeah, not sure we want this.  @vjeux?

**Comment by aleclarson** [comment]
Perhaps changing `clampValue` to `toValue`. Then, using the equation to determine minimum velocity required to travel the specified distance, we can ensure that `toValue` is reached and never exceeded. This might make the use cases more clear?

File: Libraries/Animated/Animated.js (hunk 1 above)
Comment (aleclarson) [inline Libraries/Animated/Animated.js]: Yeah I agree. Forgot to change the name after changing the function's purpose. :smile:

**Pull request closed without merging**